src/toil/job.py

```diff
@@ -1282,7 +1331,7 @@ def __init__(self, job):
         :param toil.job.Job job: the job to encapsulate.
         """
         # Giving the root of the subgraph the same resources as the first job in the subgraph.
-        Job.__init__(self, disk=job.disk, memory=job.memory, cores=job.cores)
+        Job.__init__(self, **job._resources)
```

Comment:
The tests were failing because this was previously **self._resources which made the new job always have the default requirements no matter what was passed to the constructor.